src/utils/auth_utils.js

```diff
@@ -0,0 +1,402 @@
+
+const { VOLUNTEER_ROLE } = AUTH_ROLES.AUTH_ROLES;
+
+/*
+  TODO:
+  X Login:
+    X Fix refresh token bug
+  X Get roles working:
+    X Switch from 2 booleans to string value for role
+    X Make sure correct cookies are being set
+  - Admin Invite / Forgot password page
+  X Test all CRUD routes for users
+*/
+
+// Using Firebase Web version 9
+const firebaseConfig = {
+  apiKey: process.env.REACT_APP_FIREBASE_APIKEY,
+  authDomain: process.env.REACT_APP_FIREBASE_AUTHDOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECTID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGEBUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGINGSENDERID,
+  appId: process.env.REACT_APP_FIREBASE_APPID,
+};
+
+const app = initializeApp(firebaseConfig);
+const auth = getAuth(app);
+
+// TEMP: Make sure to remove
+const NPOBackend = axios.create({
+  baseURL: 'http://localhost:3001',
+  withCredentials: true,
+});
+
+const refreshUrl = `https://securetoken.googleapis.com/v1/token?key=${process.env.REACT_APP_FIREBASE_APIKEY}`;
+
+/**
+ * Sets a cookie in the browser
+ * @param {string} key key for the cookie
+ * @param {string} value value for the cookie
+ * @param {cookieConfig} config cookie config to use
+ */
+const setCookie = (key, value, config) => {
+  let cookie = `${key}=${value}; max-age=${config.maxAge}; path=${config.path}`;
+  if (config.domain) {
+    cookie += `; domain=${config.domain}`;
+  }
+  if (config.secure) {
+    cookie += '; secure';
+  }
+  document.cookie = cookie;
+};
+
+/**
+ * Returns the current user synchronously
+ * @param {Auth} authInstance
+ * @returns The current user (or undefined)
+ */
+const getCurrentUser = authInstance =>
+  new Promise((resolve, reject) => {
+    const unsubscribe = authInstance.onAuthStateChanged(
+      user => {
+        unsubscribe();
+        resolve(user);
+      },
+      err => {
+        reject(err);
+      },
+    );
+  });
+
+// Refreshes the current user's access token by making a request to Firebase
+const refreshToken = async () => {
+  const currentUser = await getCurrentUser(auth);
+  if (currentUser) {
+    const refreshT = currentUser.refreshToken;
+    const {
+      data: { access_token: idToken },
+    } = await axios.post(refreshUrl, {
+      grant_type: 'refresh_token',
+      refresh_token: refreshT,
+    });
+    // Sets the appropriate cookies after refreshing access token
+    setCookie(cookieKeys.ACCESS_TOKEN, idToken, cookieConfig);
+    const user = await NPOBackend.get(`/users/${auth.currentUser.uid}`);
+    console.log(user.data.user);
+    setCookie(cookieKeys.ROLE, user.data.role, cookieConfig);
+    return idToken;
+  }
+  return null;
+};
+
+const sendInviteLink = async email => {
+  // deleted line where invite id was generated w/ nanoid()
+  // deleted line posting to NPOBackend (/adminInvite)
+  const inviteId = '';
+  const url = `http://localhost:3000/auth-email?mode=inviteUser&inviteID=${inviteId}`;
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'url' is assigned a value but never used.